File: JassToTs/JassException.cs
using System;

namespace Jass
{
    public class JassException : Exception
    {
        static bool isStrict = true;
        public static bool IsStrict { get => isStrict; set => isStrict = value; }
        static string formatMessage(int line, int col, string message) => $"Line {line}, Col {col}: {message}";

        public static void Error(int line, int col, string message) 
        {
            if (isStrict) throw new JassException(line, col, message);
            Console.WriteLine(formatMessage(line, col, message));
        }
        
        public JassException(int line, int col, string message) : base(formatMessage(line, col, message)) {}
    }
}

File: jass-to-ts-go/jass/JassException.go
package jass

import (
	"errors"
	"fmt"
)

var isStrict = true

func GetIsStrict() bool {
	return isStrict
}
func SetIsStrict(value bool) {
	isStrict = value
}

func formatMessage(line int, col int, message string) string {
	return fmt.Sprintf("Line %d, Col %d: %v", line, col, message)
}

func JassError(line int, col int, message string) error {
	msg := formatMessage(line, col, message)
	if isStrict {
		return errors.New(msg)
	}
	fmt.Println(msg)
	return nil
}
